packages/terra-date-time-picker/src/DateTimeUtils.js

```diff
@@ -238,12 +238,12 @@ class DateTimeUtils {
    * @param {string|undefined} date - The date to convert. Expect to be in ISO format.
    * @return {object|undefined} - The moment object. Undefined if unable to convert.
    */
-  static createSafeDate(date) {
+  static createSafeDate(date, timeZone) {
     if (!date) {
       return undefined;
     }
 
-    const momentDate = moment(date);
+    const momentDate = (timeZone && moment.tz.zone(timeZone) ? moment.tz(date, timeZone) : moment.tz(date, moment.tz.guess()));
```

Comment:
When no timezone is passed in, instead of doing`moment.tz(date, moment.tz.guess())` would it also work if we do `moment(date)` just like what it used to do?

Reply:
Yes we could. updated here 3d9c068396910f04f05e60a58f3c4a0c05532950

Reply:
moment(date) fails in a few scenarios and hence is reverted here 574bace57cdc094bdbb3007263af3506f5aa27d7